pkg/internal/netolly/flow/deduper.go

```diff
@@ -91,9 +91,8 @@ func DeduperProvider(dd *Deduper) (pipe.MiddleFunc[[]*ebpf.Record, []*ebpf.Recor
 				// Before forwarding, unset the non-common fields of deduplicate flows.
 				// These values are not relevant after deduplication and keeping them
 				// would unnecessarily increase cardinality, as they could chaotically
-				// contain the different directions and interfaces.
+				// contain the different interfaces.
 				record.Id.IfIndex = ebpf.InterfaceUnset
-				record.Id.Direction = ebpf.DirectionUnset
```

Comment:
Currently the connections can be traced at different levels:

* Via Linux Traffic Control (currently the default method)
* Via Socket C library (the part that you are fixing)

When the packets are traced via Linux Traffic Control, the direction field does not indicate the direction from an application perspective but the direction in the network interface where it was traced.

In scenarios like Kubernetes, the same packet will be traced in diverse directions of diverse interfaces. For example, a packet from a Pod A to a Pod B in the same host, will be inspected in:

* egress of A's veth
* ingress of Hosts' eth (as received from pod A)
* egress of Hosts' eth (as forwarded to pod B)
* ingress of B's veth

The deduper component aims to report only one packet instead of 4, and that's why for Traffic Control flows, the direction field needs to be still deduplicated and considered part of the flow key.

Actually, even for sockets still has sense that the direction is part of the identifier, and not part of the metrics (as it is someting that won't vary, and is part of the identification of the flow)

Reply:
Hi @mariomac 
Thanks for the clarification.
As far as I understand, beyla looks at metrics more from a global perspective. I mean, if there is communication between two Pods (same node) in k8s (as you described), beyla collects one egress row and one ingress row, instead of pair egres-ingress per Pod (dedupe mechanism) and in that way, an unset of `direction` make sense, but you loose information about the initiator. Am I right?

As you said, `direction` is not changed around the flow, but only because based on condition `src_port` > `dst_port` (socket filter case). 
Additionally, is `direction` needed in `flow_id`? I mean for correct identification.

Reply:
Yes, you loose information about the initiator. To be honest we never cared about it for network metrics (as opposition to application-level metrics).

Direction isn't probably required for identification/differentiation of a flow, as there are also the interface and the src/dst addresses to give uniqueness to that flow. I put direction in the ID part of the flow because I considered that semantically had more sense: it is not an actual metric that changes but information about where the flow was acquired in the internet device. But if you think there is a good reason to move the field to the metrics part, I'm ok as long as we fix the tests.

Reply:
When I said "fix the tests" I also mean "modify the tests to pass with the current changes" :-) If you need some help with them, just ping and I can try to submit a PR to your branch.

Reply:
Hi @mariomac 
Thanks for your support. I think I know more or less what has to be done with tests to make it run, but regarding to main thread comments, I think it is not worth continuing it in this model at this time. 
I think you / we should try to replace TC and socket filter with the mentioned tracepoint event.